vendor/github.com/mainflux/callhome/pkg/client/client.go

```diff
@@ -0,0 +1,142 @@
+package client
+
+import (
+	"bytes"
+	"context"
+	"crypto/tls"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/netip"
+	"strings"
+	"time"
+
+	mflog "github.com/mainflux/mainflux/logger"
+)
+
+const (
+	HomeUrl           = "https://deployments.mainflux.io/telemetry"
+	stopWaitTime      = 5 * time.Second
+	callHomeSleepTime = 30 * time.Minute
+	backOff           = 10 * time.Second
+	apiKey            = "77e04a7c-f207-40dd-8950-c344871fd516"
+)
+
+var ipEndpoints = []string{
+	"https://checkip.amazonaws.com/",
+	"https://ipinfo.io/ip",
+	"https://api.ipify.org/",
+}
+
+type homingService struct {
+	serviceName string
+	version     string
+	logger      mflog.Logger
+	cancel      context.CancelFunc
+	httpClient  http.Client
+}
+
+func New(svc, version string, homingLogger mflog.Logger, cancel context.CancelFunc) *homingService {
+	return &homingService{
+		serviceName: svc,
+		version:     version,
+		logger:      homingLogger,
+		cancel:      cancel,
+		httpClient:  *http.DefaultClient,
+	}
+}
+
+func (hs *homingService) CallHome(ctx context.Context) {
+	for {
+		select {
+		case <-ctx.Done():
+			hs.Stop()
+		default:
+			data := telemetryData{
+				Service:  hs.serviceName,
+				Version:  hs.version,
+				LastSeen: time.Now(),
+			}
+			for _, endpoint := range ipEndpoints {
+				ip, err := hs.getIP(endpoint)
+				if err != nil {
+					hs.logger.Warn(fmt.Sprintf("failed to get ip address with error: %v", err))
```

Comment:
We should make the message more descriptive (like `failed to obtain service public IP address for sending Minflux usage telemetry`) and make this a `Debug` log level.

Reply:
https://github.com/mainflux/callhome/pull/4

Reply:
https://github.com/mainflux/callhome/pull/4